be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -558,190 +562,316 @@ std::tuple<bool, orc::Literal> convert_to_orc_literal(const orc::Type* type, con
+            DCHECK(expr->children().size() == 1);
+            DCHECK(expr->children()[0]->is_slot_ref());
+            const auto* slot_ref = static_cast<const VSlotRef*>(expr->children()[0].get());
+            auto* slot = _tuple_descriptor->slots()[slot_ref->column_id()];
+            const auto* orc_type = _type_map[_col_name_to_file_col_name[slot->col_name()]];
+            const auto predicate_type = TYPEKIND_TO_PREDICATE_TYPE[orc_type->getKind()];
+            builder->isNull(slot_ref->expr_name(), predicate_type);
+        } else if (expr->fn().name.function_name == "is_not_null_pred") {
+            DCHECK(expr->children().size() == 1);
+            DCHECK(expr->children()[0]->is_slot_ref());
+            const auto* slot_ref = static_cast<const VSlotRef*>(expr->children()[0].get());
+            auto* slot = _tuple_descriptor->slots()[slot_ref->column_id()];
+            const auto* orc_type = _type_map[_col_name_to_file_col_name[slot->col_name()]];
+            const auto predicate_type = TYPEKIND_TO_PREDICATE_TYPE[orc_type->getKind()];
+            builder->startNot();
+            builder->isNull(slot_ref->expr_name(), predicate_type);
+            builder->end();
+        } else {
+            __builtin_unreachable();
         }
-        builder->end();
+        break;
+    }
+    default: {
+        // should not reach here, because _check_expr_can_push_down has already checked
+        __builtin_unreachable();
+    }
     }
     return true;
 }
 
-bool OrcReader::_init_search_argument(
-        std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range) {
-    if ((!_enable_filter_by_min_max) || colname_to_value_range->empty()) {
+bool OrcReader::_init_search_argument(const VExprContextSPtrs& conjuncts) {
+    if (!_enable_filter_by_min_max) {
         return false;
     }
-    std::vector<OrcPredicate> predicates;
-    auto& root_type = _reader->getType();
-    std::unordered_map<std::string, const orc::Type*> type_map;
-    for (int i = 0; i < root_type.getSubtypeCount(); ++i) {
-        type_map.emplace(get_field_name_lower_case(&root_type, i), root_type.getSubtype(i));
+
+    // if no conjuncts can be pushed down, return false
+    if (!std::ranges::any_of(conjuncts, [this](const auto& expr_ctx) {
+            return _check_expr_can_push_down(expr_ctx->root());
+        })) {
+        return false;
     }
-    for (auto& col_name : _lazy_read_ctx.all_read_columns) {
-        auto iter = colname_to_value_range->find(col_name);
-        if (iter == colname_to_value_range->end()) {
-            continue;
-        }
-        auto type_it = type_map.find(_col_name_to_file_col_name[col_name]);
-        if (type_it == type_map.end()) {
-            continue;
+
+    auto sargBuilder = orc::SearchArgumentFactory::newBuilder();
+    sargBuilder->startAnd();
+    for (const auto& expr_ctx : conjuncts) {
+        if (!_build_search_argument(expr_ctx->root(), sargBuilder)) {
```

Comment:
if there is A、B、C in `conjuncts`, and if `_build_search_argument(A)` return false, then we no longer try to push down B and C?

Reply:
Indeed a problem, return false only if all conjuncts  cannot be _build_search_argument normally.